src/main/java/com/artgallery/ArtService.java

```diff
@@ -57,6 +61,138 @@ public void buyPainting() {
         repository.addPainting(painting);
     }
 
+    public void editPainting() {
+        System.out.println("Editing painting...");
+        List<Painting> paintings = repository.getPaintings();
+        showMyCollection(paintings);
+        System.out.println("Choose Painting to edit: ");
+        int index;
+        while (true) {
+            try {
+                index = getIndex();
+                Painting painting = paintings.get(index);
+                editing(painting);
+                break;
+            } catch (InputMismatchException | IndexOutOfBoundsException exception) {
+                System.out.println("Please choose number between 1 and " + repository.getPaintings().size());
+                scanner.nextLine();
+            }
+        }
+    }
+
+    private void editing(Painting painting) {
+        System.out.println("Do you want to change data: " + "\n" + painting.getAuthor().getDescription() + "? [y]es / [n]o");
+        String optionA = getOption();
```

Comment:
optionA i optionP to może być jeden option, rozbicie na dwie zmienne nie ma sensu